### PR TITLE
fix: added check for feature conflict

### LINKED ISSUE #1487: Add new feature frequency-lint-check for feature conflict check
CI will need to pass the feature `frequency-lint-check` when running lints, checks and benchmarks.  Conflict check that produces a compile error in `node/cli/build.rs` is to be re-enabled.

### PR DESCRIPTION
**Pull request by rlaferla**
# Goal
The goal of this PR is to generate a compile error if two main features are enabled and it's not for benchmarking/lint/check.

Closes  #1487

### CODE REVIEW

File: node/service/src/lib.rs [hunk 1]
```diff
@@ -15,7 +15,7 @@ compile_error!(
 - Frequency Rococo: "frequency-rococo-testnet"
 - Local: "frequency-rococo-local"
 - No Relay: "frequency-no-relay",
-- All: "all-frequency-features""#
+- All: "frequency-lint-check"#
 );
```
Comment (wilwade) [inline node/service/src/lib.rs:19]: Should this compile error go into build.rs?

Comment (rlaferla) [reply node/service/src/lib.rs]: done

File: node/Cargo.toml [hunk 2]
```diff
@@ -38,7 +38,7 @@ frequency-rococo-testnet = [
   "frequency-service/frequency-rococo-testnet",
   "frequency-cli/frequency-rococo-testnet"
 ]
-all-frequency-features = [
-  "frequency-service/all-frequency-features",
-  "frequency-cli/all-frequency-features"
+frequency-lint-check = [
```
Comment (JoeCap08055) [inline node/Cargo.toml:41]: 💯 This naming makes so much more sense

**Review by shannonwells — APPROVED** [review]
Looks great, thanks!

File: tools/scripts/build.sh [hunk 3]
```diff
@@ -15,7 +15,7 @@ cargo --version
 
 case $TARGET in
```
Comment (wilwade) [inline tools/scripts/build.sh:16]: I don't think this file is being used anymore, but that doesn't hold up the merge.

**Review by wilwade — APPROVED** [review]
✨

**Review by mattheworris — APPROVED** [review]
🔥

File: tools/scripts/build.sh (hunk 3 above)
Comment (rlaferla) [reply tools/scripts/build.sh]: I've given it the axe.  It's a dead parrot.

File: node/cli/build.rs [hunk 4]
```diff
@@ -1,22 +1,63 @@
...
+);
 
-// Don't allow both frequency-no-relay and all-frequency-features so that we always have a good mainnet runtime
-// #[cfg(all(
-// 	feature = "frequency",
-// 	feature = "all-frequency-features",
-// 	not(feature = "runtime-benchmarks")
-// ))]
-// compile_error!("feature \"frequency-no-relay\" and feature \"all-frequency-features\" cannot be enabled at the same time");
+// Don't allow more than one main feature (except for benchmark/lint/check) so that we always have a good mainnet runtime
```
Comment (JoeCap08055) [inline node/cli/build.rs:18]: Ever look into the `cfg-if` crate?

**Review by JoeCap08055 — APPROVED** [review]

File: node/cli/build.rs (hunk 4 above)
Comment (rlaferla) [reply node/cli/build.rs:18]: Yes but trying to minimize dependencie.

**Pull request merged**